### PR TITLE
[Enhancement] Enhance FE/BE to output logs to console directly

### PR DESCRIPTION
**Pull request by kevincai**
* LOG_CONSOLE env var is checked in fe/be/cn entrypoint script and pass
  `--logconsole` parameter to start_<fe|be|cn>.sh
* start_fe.sh force set `syslog_to_console = true` in fe.conf so all
  logs are written to console stderr
* start_be.sh/start_cn.sh force set `GLOG_logtostderr=1` env var to
  redirect glog output to console stderr
* remove writing to pidfile, FE/BE application code will take care it.

Signed-off-by: Kevin Xiaohua Cai <caixiaohua@starrocks.com>

## What type of PR is this：
- [ ] BugFix
- [ ] Feature
- [X] Enhancement
- [ ] Refactor
- [ ] UT
- [ ] Doc
- [ ] Tool

## Which issues of this PR fixes ：

Fixes #

## Checklist:

- [ ] I have added test cases for my bug fix or my new feature
- [ ] This pr will affect users' behaviors
- [ ] This pr needs user documentation (for new or modified features or behaviors)
  - [ ] I have added documentation for my new feature or new function

## Bugfix cherry-pick branch check:
- [X] I have checked the version labels which the pr will be auto backported to target branch
  - [X] 3.0
  - [X] 2.5
  - [ ] 2.4
  - [ ] 2.3

### CODE REVIEW

File: bin/start_backend.sh [hunk 1]
```diff
@@ -177,9 +180,18 @@ if [ ${RUN_CN} -eq 1 ]; then
     LOG_FILE=${LOG_DIR}/cn.out
 fi
 
-echo "start time: "$(date) >> ${LOG_FILE}
+if [ ${RUN_LOGSTDOUT} -eq 1 ] ; then
+    # force glog output to console, redirect stderr to stdout
+    export GLOG_logtostderr=1
+    exec 2>&1
```
Comment (dengliu) [inline bin/start_backend.sh:186]: I think we can let error goes to `2` `stderr` so this let datadog agent to add the default status of `Error` when export to DD backend.

File: bin/start_fe.sh [hunk 2]
```diff
@@ -143,10 +151,23 @@ if [ x"$HOST_TYPE" != x"" ]; then
     HOST_TYPE="-host_type $HOST_TYPE"
 fi
 
-if [ ${RUN_DAEMON} -eq 1 ]; then
-    nohup $LIMIT $JAVA $final_java_opt com.starrocks.StarRocksFE ${HELPER} ${HOST_TYPE} "$@" >> $LOG_DIR/fe.out 2>&1 </dev/null &
+LOG_FILE=$LOG_DIR/fe.out
+
+if [ ${RUN_LOGSTDOUT} -eq 1 ] ; then
+    # redirect all subsequent commands' stderr to stdout
+    exec 2>&1
```
Comment (dengliu) [inline bin/start_fe.sh:158]: ditto

File: bin/start_backend.sh (hunk 1 above)
Comment (kevincai) [reply bin/start_backend.sh:186]: done

**Review by dengliu — APPROVED** [review]
LGTM!

**Review by imay — APPROVED** [review]

File: bin/start_fe.sh [hunk 3]
```diff
@@ -143,10 +141,28 @@ if [ x"$HOST_TYPE" != x"" ]; then
     HOST_TYPE="-host_type $HOST_TYPE"
 fi
 
-if [ ${RUN_DAEMON} -eq 1 ]; then
-    nohup $LIMIT $JAVA $final_java_opt com.starrocks.StarRocksFE ${HELPER} ${HOST_TYPE} "$@" >> $LOG_DIR/fe.out 2>&1 </dev/null &
+LOG_FILE=$LOG_DIR/fe.out
+
+if [ ${RUN_LOG_CONSOLE} -eq 1 ] ; then
+    if [ ! -w $STARROCKS_HOME/conf/fe.conf ] ; then
+        # workaround configmap readonly, can't change its content
```
Comment (chaoyli) [inline bin/start_fe.sh:148]: Why not `cp $STARROCKS_HOME/conf/fe.conf $STARROCKS_HOME/conf/fe.conf.readonly`?

Comment (kevincai) [reply bin/start_fe.sh:148]: fe.conf is a symbol link, cp fe.conf to fe.conf.readonly, doesn't make fe.conf writable.

**Review by chaoyli — APPROVED** [review]

**Comment by wanpengfei-git** [comment]
@Mergifyio backport branch-3.0

**Comment by wanpengfei-git** [comment]
@Mergifyio backport branch-2.5

**Comment by kevincai** [comment]
@mergifyio backport branch-3.0

**Pull request merged**